### PR TITLE
[WIP] Update benchmark readme

### PR DESCRIPTION
**Pull request by gottesmm**
This is a WIP to update the README to the current state of the repository.

### CODE REVIEW

**Comment by gottesmm** [comment]
@natecook1000 Can I get your opinion on this... have I made it clearer?

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by Jceciliani** [comment]
I cannot edit this file due to permissions. When trying to push to Github Desktop, there is a JS error not allowing it, any suggestions? Based on what I know, the content looks good, I just want to make some texts fixes here and there before a merge.

**Comment by gottesmm** [comment]
@Jceciliani I would just comment on the PR itself using github's review UI.

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: Recommended change to "of regression, micro, small, and medium sized benchmarks that are used"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "There are 3 ways to build the benchmark suite:"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "**NOTE:** There are various levels of support here. Specifically:"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "Non-macOS Darwin platforms today do not support the swiftpm build."

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "This is not due to intention, but rather, more engineering time would be necessary to produce these builds."

I am not entirely sure what is trying to be said here.

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "This can be used to generate an xcode project that can be used to edit the"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "**NOTE:** By default, the xcode's project scheme for SwiftBench will build the"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: I would take out "just" at the end.

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "to be built" or "to build"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "SwiftBench will be a listed built product in Xcode. To find the path of"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "in the" at the end

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: sample, the test runs are significantly longer than system jitter. Due to this, we can"

File: benchmark/README.md
Comment (Jceciliani) [inline benchmark/README.md]: "This is done by the low level benchmark driver. It will measure the time taken for"

**Review by Jceciliani — CHANGES_REQUESTED** [review]
These are a few of the text changes I would recommend for this file.

File: benchmark/README.md
Comment (palimondo) [inline benchmark/README.md]: How does one control the optimization level when building with `swiftpm`? Maybe an example here would help?

**Comment by gottesmm** [comment]
Thanks! Let me take a look at this.

File: benchmark/README.md
Comment (palimondo) [inline benchmark/README.md]: Non-unstable? As in stable? Or maybe “Run all benchmarks (except those tagged `.unstable`) for 3 samples…”

**Review by palimondo — CHANGES_REQUESTED** [review]
I got these suggested  changes. (I hope I’m using this review interface right…)

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: "productization" maybe is something that we need to add to the lexicon. Basically the idea is that there are certain things that are "possible" to do with this software today that may either be fragile, unsupported, or not tested. Productization is changing the software so that it is easily usable by others.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: The apostrophe here is correct. I am not talking about xcode (the app)'s project scheme. I am talking about the "xcode project's" scheme. The spelling error is correct here.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: That is the intention behind why we do this, so it is appropriate. If/when we make any of the proposed changes, we can change this.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: swiftpm isn't a command. But I am fine with changing the rest.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: This is on purpose. Iterations is an expert level feature. It confuses people as a concept and should only be progressively disclosed if necessary. This README is the right place to do that.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: Thats reasonable.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: I am also fine with this.

File: benchmark/README.md
Comment (palimondo) [reply benchmark/README.md]: I’m confused, did you mean “This README **isn’t** the right place…”?

Comment (gottesmm) [reply benchmark/README.md]: Correct.

**Comment by gottesmm** [comment]
Pavol, I am not going to add the rest of that stuff. If you think it is important create a PR and it can be talked about there. I only have a limited amount of time to spend on this.

**Comment by gottesmm** [comment]
*and I need to move on. I am just waiting on some feedback from @natecook1000. But I don't want to add more content to this PR.

File: benchmark/README.md
Comment (palimondo) [inline benchmark/README.md]: Typo: "aretwo" is missing a space

File: benchmark/README.md
Comment (palimondo) [inline benchmark/README.md]: This one didn't work for me. I've set the both variables to point to the corresponding build directory, but the `LLVM_BUILD_DIR` isn't being picked up correctly:
````
CMake Error at cmake/modules/SwiftBenchmarkUtils.cmake:61 (message):
  Unable to find Clang driver
Call Stack (most recent call first):
  cmake/modules/AddSwiftBenchmarkSuite.cmake:27 (runcmd)
  CMakeLists.txt:254 (configure_build)
````
which seem's to try to infer the CLANG_EXEC again from SWIFT_DARWIN_XCRUN_TOOLCHAIN. I think there's bug in AddSwiftBenchmarkSuite.cmake. Line 24 should IMO include `if(NOT CLANG_EXEC)` check similar the the `if(NOT SWIFT_EXEC)` on line 14.

File: benchmark/README.md
Comment (gottesmm) [reply benchmark/README.md]: Thanks.

**Comment by gottesmm** [comment]
I need to finish some stuff up today, but I am going to try and swing back around to this later today.

**Comment by gottesmm** [comment]
@palimondo realistically I am not going to have time to do this. If you want to do something like this go ahead. = ).

**Comment by palimondo** [comment]
@gottesmm Can you restore this branch an let me make a fork, please?

**Comment by palimondo** [comment]
@gottesmm Nevermind. I have grabbed the raw of the modified README.md. I’m good. I’ll have a look at updating and simplifying it based on this excellent rewite of yours, eventually. Thank you!

**Pull request closed without merging**
